nearcore/src/config.rs

```diff
@@ -1415,3 +1423,73 @@ fn test_config_from_file() {
         );
     }
 }
+
+#[test]
+fn test_create_testnet_configs() {
+    let num_shards = 4;
+    let num_validator_seats = 4;
+    let num_non_validator_seats = 8;
+    let prefix = "node";
+    let local_ports = true;
+
+    // Set all supported options to true and verify config and genesis.
+
+    let archive = true;
+    let fixed_shards = true;
+    let track_all_shards = true;
+
+    let (configs, _validator_signers, _network_signers, genesis, _shard_keys) =
+        create_testnet_configs(
+            num_shards,
+            num_validator_seats,
+            num_non_validator_seats,
+            prefix,
+            local_ports,
+            archive,
+            fixed_shards,
+            track_all_shards,
+        );
+
+    assert_eq!(
+        u64::try_from(configs.len()).unwrap(),
+        num_validator_seats + num_non_validator_seats
+    );
+
+    for config in configs {
+        assert_eq!(config.archive, true);
+        assert_eq!(config.tracked_shards.len(), 1);
+    }
+
+    assert_eq!(genesis.config.validators.len(), 4);
+    assert_eq!(genesis.config.shard_layout.num_shards(), 4);
+
+    // Set all supported options to false and verify config and genesis.
+
+    let archive = false;
+    let fixed_shards = false;
+    let track_all_shards = false;
+
+    let (configs, _validator_signers, _network_signers, genesis, _shard_keys) =
+        create_testnet_configs(
+            num_shards,
+            num_validator_seats,
+            num_non_validator_seats,
+            prefix,
+            local_ports,
+            archive,
+            fixed_shards,
+            track_all_shards,
+        );
+    assert_eq!(
+        u64::try_from(configs.len()).unwrap(),
+        num_validator_seats + num_non_validator_seats
+    );
+
+    for config in configs {
+        assert_eq!(config.archive, false);
+        assert_eq!(config.tracked_shards.len(), 0);
+    }
+
+    assert_eq!(genesis.config.validators.len(), 4);
+    assert_eq!(genesis.config.shard_layout.num_shards(), 4);
```

Comment:
Please create a `NUM_SHARDS` constant instead of repeating the value `4` in this function.

Reply:
done